src/EnergyPlus/OutputProcessor.cc

```diff
@@ -136,6 +137,49 @@ namespace OutputProcessor {
         state.dataOutputProcessor->IVariableTypes.redimension(state.dataOutputProcessor->MaxIVariable += IVarAllocInc);
     }
 
+    int DetermineMinuteForReporting(EnergyPlusData &state, TimeStepType t_timeStepType) // kind of reporting, Zone Timestep or System
+    {
+
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Linda Lawrie
+        //       DATE WRITTEN   January 2012
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // When reporting peaks, minutes are used but not necessarily easily calculated.
+
+        // METHODOLOGY EMPLOYED:
+        // Could use the access to the minute as OP (OutputProcessor) does but uses
+        // external calculation.
+
+        // Using/Aliasing
+        auto &SysTimeElapsed = state.dataHVACGlobal->SysTimeElapsed;
+        auto &TimeStepSys = state.dataHVACGlobal->TimeStepSys;
+
+        // Return value
+        int ActualTimeMin; // calculated Minute for reporting
+
+        // FUNCTION PARAMETER DEFINITIONS:
+        Real64 constexpr FracToMin(60.0);
+
+        // FUNCTION LOCAL VARIABLE DECLARATIONS:
+        Real64 ActualTimeS; // Start of current interval (HVAC time step)
+        Real64 ActualTimeE; // End of current interval (HVAC time step)
+        int ActualTimeHrS;
+
+        if (t_timeStepType == OutputProcessor::TimeStepType::System) {
+            ActualTimeS = state.dataGlobal->CurrentTime - state.dataGlobal->TimeStepZone + SysTimeElapsed;
+            ActualTimeE = ActualTimeS + TimeStepSys;
+            ActualTimeHrS = int(ActualTimeS);
+            ActualTimeMin = nint((ActualTimeE - ActualTimeHrS) * FracToMin);
+        } else {
+            ActualTimeMin = (state.dataGlobal->CurrentTime - int(state.dataGlobal->CurrentTime)) * FracToMin;
+        }
```

Comment:
Thanks @rraustad, note taken.  I definitely didn't look at that level, but it is certainly possible (likely?) that this can be trimmed down.  I'm open to trying the change locally to see if it causes regressions.

Reply:
See the difference between GetCurrentHVACTime and GetPreviousHVACTime

Reply:
I took a minute and used your one-line expression for this function and :magic_wand:  magic :magic_wand: , no test failures or diffs.  Thanks for the notable simplification!

I am pushing that change up, along with addressing the other minor things mentioned in previous comments shortly.  I think I am done working on this branch, assuming CI is happy, and moving back to other tasks.  Any additional comments are welcomed, but will by default be :pushpin: for later.

Reply:
As long as we have results from a detailed time step simulation to comapre to.

Reply:
@rraustad so I have a bit more confidence.  Our test suite has almost 600 output variables declared as "detailed" across 42 files.  (From a reasonable grep.)  I see no issues in regressions or unit test failures with the change.

In addition, and maybe more importantly, our CI dashboard is showing that both sides of the IF block are getting hit many, many times during our unit test and integration test runs.

Unit test coverage [link](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2023-01-27-EnergyPlus-1504f8b59b-x86_64-gcc-7.5UTCD/EnergyPlus/General.cc.gcov.html):

![image](https://user-images.githubusercontent.com/849824/216151883-b8d6f2e0-8ea8-4189-8d41-b9c2ca168e3f.png)

Integration test coverage [link](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2023-01-27-EnergyPlus-1504f8b59b-x86_64-gcc-7.5ICD/EnergyPlus/General.cc.gcov.html):

![image](https://user-images.githubusercontent.com/849824/216151973-69a75be7-9b61-4b47-af16-3d7be3328b4a.png)

Note that these results are from develop, where the file lived in General.cc, because that's the easiest way to see both sides of the IF block prior to this change.

Anyway, after seeing that code is hit thousands of times in our test suite, and we are seeing no issue, I'm inclined to continue.  Anyone have any issue with it?

Reply:
I've thought about that equation and it looks correct to me. If there are no diffs then just like any other bug, we would wait for a future issue. Forge ahead.